src/akkudoktoreos/utils/datetimeutil.py

```diff
@@ -250,73 +249,271 @@ def to_timedelta(
         matches = re.findall(r"(\d+)\s*(days?|hours?|minutes?|seconds?)", input_value)
 
         if not matches:
-            raise ValueError(f"Invalid time string format: {input_value}")
+            error_msg = f"Invalid time string format '{input_value}'"
+            logger.error(error_msg)
+            raise ValueError(error_msg)
 
         for value, unit in matches:
             unit = unit.lower().rstrip("s")  # Normalize unit
             if unit in time_units:
                 total_seconds += int(value) * time_units[unit]
             else:
-                raise ValueError(f"Unsupported time unit: {unit}")
+                error_msg = f"Unsupported time unit: {unit}"
+                logger.error(error_msg)
+                raise ValueError(error_msg)
 
-        return timedelta(seconds=total_seconds)
+        return pendulum.duration(seconds=total_seconds)
 
     else:
-        raise ValueError(f"Unsupported input type: {type(input_value)}")
-
+        error_msg = f"Unsupported input type: {type(input_value)}"
+        logger.error(error_msg)
+        raise ValueError(error_msg)
 
-@overload
-def to_timezone(lat: float, lon: float, as_string: Literal[True]) -> str: ...
 
+timezone_finder = TimezoneFinder()  # Static variable for caching
 
-@overload
-def to_timezone(lat: float, lon: float, as_string: Literal[False] | None = None) -> ZoneInfo: ...
 
+def to_timezone(
+    utc_offset: Optional[float] = None,
+    location: Optional[Set[float]] = None,
```

Comment:
should probably be Optional[tuple[float, float]]

Reply:
Changed to Tuple.